crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1242,12 +1248,16 @@ impl<'db> TypeInferenceBuilder<'db> {
         Type::BooleanLiteral(*value)
     }
 
-    #[allow(clippy::unused_self)]
     fn infer_string_literal_expression(&mut self, literal: &ast::ExprStringLiteral) -> Type<'db> {
-        Type::StringLiteral(StringLiteralType::new(self.db, literal.value.to_string()))
+        let value = if literal.value.len() <= Self::MAX_STRING_LITERAL_SIZE {
+            literal.value.to_str().into()
+        } else {
+            Box::default()
```

Comment:
This seems problematic because we now have no way of knowing whether we omitted a string literal type from inference or if it is an empty string in the source code.

Reply:
Interesting — what situation do you have in mind where we might need to distinguish between the two?

Reply:
For example in `a + b` we concatenated the strings. If `a > MAX_STRING_LITERAL_SIZE` then the concatenated string is just `b` but that's not the correct literal value. 

I also think it's relevant when displaying the type. We should display a "truncated" string type as `Literal[str]` and not `Literal[""]`

Reply:
Yeah sorry I misread and deleted my comment